python-threatexchange/threatexchange/signal_type/tests/test_hash_from_x.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import unittest
+import os
+
+from threatexchange.content_type import meta
+from threatexchange.signal_type.signal_base import FileHasher, StrHasher
+from pathlib import Path
+
+
+class TestHashFrom(unittest.TestCase):
+    """
+    Signal Type has a weird inheritance structure the error around which are easy to
+    miss before run time. If a SignalType is a FileHasher/StrHasher it should be able to
+    take the corresponding content medium without throwing.
+    """
+
+    def test_file_hashers_have_impl(self):
+        signal_types = [s() for s in meta.get_all_signal_types()]
+        file_hashers = [s for s in signal_types if issubclass(type(s), FileHasher)]
+        for file_hasher in file_hashers:
+            file_hasher.hash_from_file(
+                Path(os.path.dirname(__file__) + "/../../../tests/data/b.jpg")
```

Comment:
I think this might have issues if run from other directories, or if we move this code later, but I don't know how to fix it

Reply:
that is why I take the dirname of the file, it should only break if either the test or the image file move

Reply:
If you only just need a file to exist (since you aren't testing a specific file), check out the tmpfile: https://docs.python.org/3/library/tempfile.html